### PR TITLE
libarchive: fix wrong functions checks

### PR DESCRIPTION
**Pull request by neheb**
I removed these upstream in order to use the macros in the headers.

### CODE REVIEW

**Comment by benoit-pierre** [comment]
Meaning?

**Comment by benoit-pierre** [comment]
`HAVE__MKGMTIME` is still used, for example.

**Comment by benoit-pierre** [comment]
Wait, never mind, I'm stupid and/or too tired…

File: subprojects/packagefiles/libarchive/meson.build [hunk 1]
```diff
@@ -107,6 +107,12 @@ if cc.has_header_symbol('errno.h', 'EILSEQ')
   cdata.set('HAVE_EILSEQ', 1)
 endif
 
+foreach w : ['_mkgmtime', 'ctime_s', 'gmtime_s', 'localtime_s']
+  if cc.has_header_symbol('time.h', w)
+    cdata.set('HAVE_@0@'.format(w.to_upper()), 1)
+  endif
+endforeach
+
```
Comment (benoit-pierre) [inline subprojects/packagefiles/libarchive/meson.build:115]: Maybe move the check for `gmtime_r` here too?

Comment (neheb) [reply subprojects/packagefiles/libarchive/meson.build:115]: No need. I moved the code upstream to try the _s versions first and then the _r. _r is implemented in terms of _s as a macro IIRC. _r on POSIX systems is available as a function.

**Comment by neheb** [comment]
Reference: https://github.com/libarchive/libarchive/commits?author=neheb

**Comment by neheb** [comment]
Added missing __MINGW_USE_VC2005_COMPAT. We don't test MinGW32 but might as well match upstream.

**Pull request merged**